pkg/block/metadata/deletionmark.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metadata
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	"github.com/go-kit/kit/log"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+const (
+	// DeletionMarkFilename is the known json filename to store details about when block is marked for deletion.
+	DeletionMarkFilename = "deletion-mark.json"
+
+	// DeletionMarkVersion1 is the version of deletion-mark file supported by Thanos.
+	DeletionMarkVersion1 = 1
+)
+
+// ErrorDeletionMarkNotFound is the error when deletion-mark.json file is not found.
+var ErrorDeletionMarkNotFound = errors.New("deletion-mark.json not found")
+
+// ErrorUnmarshalDeletionMark is type wrapper for errors to indicate that deletion-mark.json file cannot be unmarshalled.
+// This can occur because deletion-mark.json file has been partially uploaded and thus cannot be unmarshalled.
+type ErrorUnmarshalDeletionMark struct {
+	path string
```

Comment:
Let's remove this  - and instead wrap it as we do below, WDYT?

Reply:
So we can have just sentinel:

```
var ErrorUnmarshalDeletionMark = errors.New(".......")
```

Reply:
Yes, I can make this change. We need not create a new Error type for this.

Reply:
I have added this as:

```
// ErrorUnmarshalDeletionMark is the error when unmarshalling deletion-mark.json file.
// This error can occur because deletion-mark.json has been partially uploaded to block storage
// or the deletion-mark.json file is not a valid json file.
var ErrorUnmarshalDeletionMark = errors.New("unmarshal deletion-mark.json")
```